File: api/Nathalie.Registry/Nathalie.Registry.DataLayer/Models/RegistryEntityRow.cs
using System.Collections.Generic;
using System.Linq;
using Nathalie.Registry.DataLayer.Sys.Attributes;

namespace Nathalie.Registry.DataLayer.Models
{
    [MongoCollection("registryData")]
    public class RegistryEntityRow : DocumentBase
    {
        public int RowNumber { get; set; }

        public List<RegistryEntityRowField> ColumnValues { get; set; }

        public bool IsEmpty()
        {
            return ColumnValues?.Where(item => item.Value != null).Any() != true;
        }
    }
}

File: app/nathalie-registry/src/app/models/registry-entity-row.ts
import {RegistryEntityRowField} from "./registry-entity-row-field";
import {KeyValue} from "./key.value";
import {TemplateField} from "./template.field";

export class RegistryEntityRow {
  rowNumber: number = 1;
  columnValues: Array<RegistryEntityRowField> = [];
}
